cli/pkg/kctrl/cmd/package/init.go

```diff
@@ -64,54 +64,63 @@ func (o *InitOptions) Run() error {
 		}
 	}
 
+	o.ui.PrintInformationalText("\nWelcome! Before we start, do install the latest Carvel suite of tools, specifically ytt, imgpkg, vendir and kbld as these will be used by kctrl.\n")
+	o.ui.PrintHeaderText("\nBasic Information")
+
 	pkgBuild, err := o.newOrExistingPackageBuild()
 	if err != nil {
 		return err
 	}
 
 	pkg, pkgMetadata, pkgInstall, err := o.newOrExistingPackageResources()
 
-	o.ui.PrintInformationalText("\nWelcome! Before we start, do install the latest Carvel suite of tools, specifically ytt, imgpkg, vendir and kbld as these will be used by kctrl.\n")
-
-	o.ui.PrintHeaderText("\nBasic Information")
-
 	pkgRefName, err := o.readPackageRefName(pkgMetadata.Name)
 	if err != nil {
 		return err
 	}
 
-	// TODO: @praveenrewar Can this be don any better?
-	pkgMetadata.Name = pkgRefName
-	pkg.Spec.RefName = pkgRefName
-	pkgMetadata.Spec.DisplayName = strings.Split(pkgRefName, ".")[0]
+	o.updateAndDefaultPackageResources(pkgRefName, pkgMetadata, pkg)
 
-	shortDesc := pkgMetadata.Spec.ShortDescription
-	if len(shortDesc) == 0 {
-		pkgMetadata.Spec.ShortDescription = pkgRefName
+	err = o.savePackageResources(pkg, pkgMetadata, pkgInstall)
+	if err != nil {
+		return err
 	}
-
-	longDesc := pkgMetadata.Spec.LongDescription
-	if len(longDesc) == 0 {
-		pkgMetadata.Spec.LongDescription = pkgRefName
+	err = pkgBuild.Save()
+	if err != nil {
+		return err
 	}
 
-	err = o.SavePackageResources(pkg, pkgMetadata, pkgInstall)
+	fetchMode, sourceConfiguration, err := sources.NewFetchConfiguration(o.ui, pkgBuild).Configure()
 	if err != nil {
 		return err
 	}
-	pkgBuild.Save()
+
+	// Source does not need to be conifgured if manifests are in the local directory
+	if sourceConfiguration != nil {
+		err = sourceConfiguration.Configure()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Should this be directly handled in the main `Configure` function?

Reply:
We will get a nil pointer exception 😢

Reply:
Before it reaches the function

Reply:
I am not sure how that will result in an exception. What I am suggesting is that instead of returning the sourceConfiguration, we directly call the sourceConfiguration.Configure in the main configure function based on the fetch option selected and then return the final result (an error).

Reply:
Oh I thought the you meant the function should handle the nil check we are doing

That is what we were doing before, It will also eliminate the need of having an interface, happy to do that 
Will make that change tomorrow